File: golang/buy_sell_stock.go
func maxProfit(prices []int) int {
	profit := 0
	min := prices[0]
	for i := 1; i < len(prices); i++ {
		if prices[i]-min > profit {
			profit = prices[i] - min
		}
		if prices[i] < min {
			min = prices[i]
		}
	}
	return profit
}

File: typescript/buy_sell_stock.ts
function maxProfit(prices: number[]): number {
	let profit = 0
	let min = prices[0]

	for (let i = 1; i < prices.length; i++) {
		let diff = prices[i] - min
		if (diff > profit) profit = diff
		if (prices[i] < min) {
			min = prices[i]
		}
	}

	return profit
}
